src/RewardsDistribution.t.sol

```diff
@@ -0,0 +1,253 @@
+        newVestId = _createVest(usr, tot2, bgn, tau, eta, mgr);
+        dist.file("vestId", newVestId);
+
+        vm.expectRevert("RewardsDistribution/empty-vest");
+        dist.distribute();
+
+        // After the initial time + the duration pass...
+        vm.warp(bgn + tau);
+
+        // We can claim the total amount `tot2` all at once.
+        vm.expectEmit(false, false, false, true, address(farm));
+        emit RewardAdded(tot2);
+        dist.distribute();
+
+        assertEq(token.balanceOf(address(farm)), totalRewards + tot + tot2);
+    }
+
+    function testRevertDistributeInvalidVestId() public {
+        // We're `file`ing a valid `vestId` on `setUp`, so we need to revert it to its initial value
+        stdstore.target(address(dist)).sig("vestId()").checked_write(bytes32(0));
+
+        vm.expectRevert("RewardsDistribution/invalid-vest-id");
+        dist.distribute();
+    }
+
+    function testRevertDistributeNoVestedAmount() public {
+        vm.expectRevert("RewardsDistribution/empty-vest");
+        dist.distribute();
+    }
+
+    function testRevertDistributeNoPendingDistributionAmount() public {
+        dist.file("calc", address(new MockZeroDistribution()));
+
+        skip(duration / 3);
+
+        vm.expectRevert("RewardsDistribution/no-pending-amount");
+        dist.distribute();
+    }
+
+    function testRevertFileInvalidVestId() public {
+        vm.expectRevert("RewardsDistribution/invalid-vest-id");
+        dist.file("vestId", 100);
+    }
+
+    function testRevertFileInvalidUsr() public {
+        address usr = address(0x1337);
+        uint256 tot = totalRewards;
+        uint256 bgn = block.timestamp; // start immediately
+        uint256 tau = duration; // 1 year duration
+        uint256 eta = 0; // No cliff; start immediatebly
+        address mgr = address(dist);
+        uint256 newVestId = _createVest(usr, tot, bgn, tau, eta, mgr);
+
+        vm.expectRevert("RewardsDistribution/invalid-vest-usr");
+        dist.file("vestId", newVestId);
+    }
+
+    function testRevertFileInvalidMgr() public {
+        address usr = address(farm);
+        uint256 tot = totalRewards;
+        uint256 bgn = block.timestamp; // start immediately
+        uint256 tau = duration; // 1 year duration
+        uint256 eta = 0; // No cliff; start immediatebly
+        address mgr = address(0x1337);
+        uint256 newVestId = _createVest(usr, tot, bgn, tau, eta, mgr);
+
+        vm.expectRevert("RewardsDistribution/invalid-vest-mgr");
+        dist.file("vestId", newVestId);
+    }
+
+    function testAuth() public {
+        checkAuth(address(dist), "RewardsDistribution");
+    }
+
+    function testFile() public {
+        // `checkFileUint` increaments the current value of the param being modified. Since `vestId` is validated,
+        // we need to create a new one to make sure the `file`d param is valid.
+        _createVest();
+        checkFileUint(address(dist), "RewardsDistribution", ["vestId"]);
```

Comment:
Would add a check here that `lastVestedAt` is set to 0 as well.

Reply:
Fixed.

Reply:
Fixed. Can't resolve comments.